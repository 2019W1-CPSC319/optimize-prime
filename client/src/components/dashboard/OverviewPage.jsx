import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import * as user from "../../actions/userActions";
import { withStyles } from '@material-ui/core/styles';
import clsx from 'clsx';

const styles = {
  title: {
    fontWeight: 'normal',
    marginLeft: '30px',
  },
}

class OverviewPage extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    const { classes, ...userProfile } = this.props
    return (
      <div>
        <div className={clsx(classes.header, classes.flex)}>
          {userProfile && <h1 className={classes.title}>Welcome, {userProfile.givenName}</h1>}
        </div>
      </div>
    );
  };
};

export default withStyles(styles)(connect((state) => ({
  userProfile: user.getUserProfile(state),
  loading: user.isLoading(state),
  hasTriedLogin: user.hasTriedLogin(state)
}), (dispatch) => ({
  fetchUserProfile: () => dispatch(user.fetchUser())
}))(OverviewPage));
