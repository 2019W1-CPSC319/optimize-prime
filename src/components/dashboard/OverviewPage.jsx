import React, { Component } from 'react';

class OverviewPage extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    return <div>OVERVIEW PAGE</div>;
  }
}

export default OverviewPage;
