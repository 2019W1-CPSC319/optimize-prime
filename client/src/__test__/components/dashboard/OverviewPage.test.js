import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import OverviewPage from '../../../components/dashboard/OverviewPage';

describe('Overview Page - Render', () => {
  let wrapper;
  const classes = {};
  const actions = {};

  beforeEach(() => {
    wrapper = shallow(<OverviewPage
      classes={classes}
      actions={actions}
    />);
  });

  it('render the component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
