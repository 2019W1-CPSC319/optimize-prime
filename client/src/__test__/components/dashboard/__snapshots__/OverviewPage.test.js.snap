// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Page - Render match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
